File: data/src/main/java/com/ykb/app/cryptotrader/data/model/Parameter.java
package com.ykb.app.cryptotrader.data.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.security.core.GrantedAuthority;

@Entity
@Table(name = "PARAMETER")
@Data
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class Parameter extends BaseEntity {

    @Id
    @Column(name = "KEY", columnDefinition = "VARCHAR(256)")
    private String key;

    @Column(name = "VALUE", columnDefinition = "VARCHAR(512)")
    private String value;

    @Column(name = "TYPE", columnDefinition = "VARCHAR(32)")
    @Enumerated(EnumType.STRING)
    private ParameterType type;

    public enum ParameterType {
        SYSTEM_CONFIG, DOMAIN_VALUE
    }

}

File: library/data/data-core/src/main/kotlin/com/ykb/app/cryptotrader/data/core/model/Parameter.kt
package com.ykb.app.cryptotrader.data.core.model

import com.ykb.app.cryptotrader.data.base.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "PARAMETER")
class Parameter(
    @Id
    @Column(columnDefinition = "VARCHAR(128)")
    private val name: String,
    @Column(columnDefinition = "VARCHAR(512)")
    private val value: String,
) : BaseEntity() {

    fun getValue(): String = value
    fun getName(): String = name

}